### PR TITLE
[Fix/Inference]Fix vllm benchmark

### PR DESCRIPTION
**Pull request by yuehuayingxueluo**
## 📌 Checklist before creating the PR

- [ ] I have created an issue for this PR for traceability
- [x] The title follows the standard format: `[doc/gemini/tensor/...]: A concise description`
- [x] I have added relevant tags if possible for us to better distinguish different PRs
- [x] I have installed pre-commit: `pip install pre-commit && pre-commit install`

## 🚨 Issue number

> Link this PR to your issue with words like fixed to automatically close the linked issue upon merge
>
> e.g. `fixed #1234`, `closed #1234`, `resolved #1234`

## 📝 What does this PR do?

> Summarize your work here.
> if you have any plots/diagrams/screenshots/tables, please attach them here.

## 💥 Checklist before requesting a review

- [ ] I have linked my PR to an issue ([instruction](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue))
- [x] My issue clearly describes the problem/feature/proposal, with diagrams/charts/table/code if possible
- [x] I have performed a self-review of my code
- [x] I have added thorough tests.
- [x] I have added docstrings for all the functions/methods I implemented

## ⭐️ Do you enjoy contributing to Colossal-AI?

- [x] 🌝 Yes, I do.
- [ ] 🌚 No, I don't.

Tell us more if you don't enjoy contributing to Colossal-AI.

### CODE REVIEW

File: examples/inference/benchmark_llama.py [hunk 1]
```diff
@@ -140,11 +147,6 @@ def benchmark_inference(args):
         if args.mode == "colossalai" or args.mode == "vllm":
             data = data.tolist()
 
-        generation_config = GenerationConfig(
-            pad_token_id=tokenizer.pad_token_id,
-            max_new_tokens=args.output_len,
-        )
-
```
Comment (yuanheng-zhao) [inline examples/inference/benchmark_llama.py:147]: You might want to refer to https://github.com/hpcaitech/ColossalAI/pull/5629 to revise the generation config (Just rebase)

File: examples/inference/benchmark_llama.py [hunk 2]
```diff
@@ -89,20 +89,27 @@ def benchmark_inference(args):
...
-        model = model.eval()
+        if args.mode != "vllm":
+            if args.test_random_weight:
+                model = transformers.LlamaForCausalLM(config).cuda()
+                tokenizer = AutoTokenizer.from_pretrained("hf-internal-testing/llama-tokenizer")
+            else:
+                assert args.model_path, "When testing pretrained weights, the model path must be provided.'"
+                model = transformers.LlamaForCausalLM.from_pretrained(args.model_path).cuda()
+                tokenizer = AutoTokenizer.from_pretrained("hf-internal-testing/llama-tokenizer")
+
```
Comment (yuanheng-zhao) [inline examples/inference/benchmark_llama.py:101]: Why using condition like this?

File: examples/inference/benchmark_llama.py [hunk 3]
```diff
@@ -89,20 +89,27 @@ def benchmark_inference(args):
...
-        if args.test_random_weight:
-            model = transformers.LlamaForCausalLM(config).cuda()
-            tokenizer = AutoTokenizer.from_pretrained("hf-internal-testing/llama-tokenizer")
-        else:
-            assert args.model_path, "When testing pretrained weights, the model path must be provided.'"
-            model = transformers.LlamaForCausalLM.from_pretrained(args.model_path).cuda()
-            tokenizer = AutoTokenizer.from_pretrained("hf-internal-testing/llama-tokenizer")
 
-        model = model.eval()
+        if args.mode != "vllm":
```
Comment (yuanheng-zhao) [inline examples/inference/benchmark_llama.py:93]: It seems that you disable testing random weights on vllm? Is that a consequence of updates of vllm?

File: examples/inference/benchmark_llama.py (hunk 2 above)
Comment (yuehuayingxueluo) [reply examples/inference/benchmark_llama.py:101]: vllm will loads weights by itself, and the additional external weight loading operation here will occupy GPU memory, leading to Out of Memory (OOM).

File: examples/inference/benchmark_llama.py (hunk 3 above)
Comment (yuehuayingxueluo) [reply examples/inference/benchmark_llama.py:93]: vllm will load the weights based on the model path. Therefore, vllm does not require any additional weight loading operations.

**Review by yuanheng-zhao — APPROVED** [review]

**Pull request merged**